SendGrid/SendGrid/Resources/APIKeys.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Net.Http;
+
+namespace SendGrid.Resources
+{
+    public class APIKeysData
+    {
+        public string name { get; set; }
+    }
+
+    public class APIKeys
+    {
+        private string _endpoint;
+        private Client _client;
```

Comment:
Not necessarily, I think I was just confused from a design/composition point of view.  But I get what you are doing and it makes sense to me.  I don't think any further change is necessary.